### PR TITLE
Feature discover sitemap

### PR DESCRIPTION
**Pull request by Pouyanpi**
Hello,

Implements #3149 . Just in case someone finds it useful. 

Best,
Pi

### CODE REVIEW

File: langchain/document_loaders/sitemap.py [hunk 1]
```diff
@@ -39,6 +41,52 @@ def __init__(
...
+            + "://"
+            + f"{urlparse(self.web_path).netloc}"
+        )
+        return base_url
+
+    def _find_sitemap_in_robotstxt(self) -> List[str]:
+        import requests
+
+        """Find sitemap in robots.txt."""
+        sitemap_urls: List[str] = []
```
Comment (vempaliakhil96) [inline langchain/document_loaders/sitemap.py:61]: @Pouyanpi This should be `site_map_urls`

Comment (Pouyanpi) [reply langchain/document_loaders/sitemap.py:61]: Thanks for the suggestion. I used it as "sitemap" and is the more commonly used spelling in the context of website architecture and organization, compared to "site map." As a result, using "sitemap_urls" as the variable name might be more fitting and in line with convention. What do you think? Of course, the project's convention has to be followed.

Comment (vempaliakhil96) [reply langchain/document_loaders/sitemap.py:61]: Hey sorry for replying so late, actually I was just pointing out a mistyped variable, if you see in this method below line 65 references `site_map_urls` whereas `sitemap_urls` was declared. I think `sitemap_urls` is a more fitting name for sure.

File: langchain/document_loaders/sitemap.py [hunk 2]
```diff
@@ -39,6 +41,52 @@ def __init__(
...
+        robots_txt_url = self._base_url + "/robots.txt"
+        response = requests.get(robots_txt_url)
+        if response.status_code == 200:
+            site_map_urls = re.findall(r"Sitemap: (\S+)", response.text, re.IGNORECASE)
```
Comment (vempaliakhil96) [inline langchain/document_loaders/sitemap.py:65]: look here.

**Comment by leo-gan** [comment]
@Pouyanpi Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**